planner/core/common_plans.go

```diff
@@ -297,9 +303,32 @@ func (e *Execute) OptimizePreparedPlan(ctx context.Context, sctx sessionctx.Cont
 	}
 	e.SnapshotTS = snapshotTS
 	e.Stmt = prepared.Stmt
+	e.handleStalenessOption(sctx)
 	return nil
 }
 
+func (e *Execute) handleStalenessOption(sctx sessionctx.Context) {
```

Comment:
You should move forward the handle of `read_ts`. Before `prepared.SchemaVersion != is.SchemaMetaVersion() {`, setting `snapshotTS`, `txnScope` and `is` rather than directly setting `e.xxx`.

The infoschema check should not be skipped.

Reply:
updated.